From b7a40f3f3e52263e74bb8409bcc89ece410225b5 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 12 Oct 2020 11:03:48 +0530
Subject: [PATCH 656/767] octeontx2-af: Fix loading external MKEX profile

Loading MKEX profile present either in KPU profile
from firmware interface or from ATF will succeed only
when Parse Nibble extraction is valid. But current
code switches to new MKEX when Parse Nibble
extraction is NOT valid by mistake hence fix it.

Fixes: 4a4baa4d("octeontx2-af: prepare for custom KPU")
Change-Id: Ied49b4afa79ab37949c14b93731c74fedade5232
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37728
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index fa288c544a33..6305f31c469e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -1145,7 +1145,7 @@ static void npc_load_mkex_profile(struct rvu *rvu, int blkaddr,
 		if (mcam_kex->mkex_sign == MKEX_SIGN &&
 		    !strncmp(mcam_kex->name, mkex_profile, MKEX_NAME_LEN)) {
 			/* If profile is valid, switch to it. */
-			if (!is_parse_nibble_config_valid(rvu, mcam_kex))
+			if (is_parse_nibble_config_valid(rvu, mcam_kex))
 				rvu->kpu.mkex = mcam_kex;
 			goto program_mkex;
 		}
-- 
2.31.1

